fe/fe-core/src/main/java/com/starrocks/privilege/PEntryObject.java

```diff
@@ -2,43 +2,51 @@
 
 package com.starrocks.privilege;
 
-import com.google.gson.annotations.SerializedName;
 import com.starrocks.server.GlobalStateMgr;
 
-import java.util.Objects;
-
 /**
- * interface is hard to serialized/deserialized using GSON
- * to simplify the implementation, the base class PEntryObject contains one data field called `id`.
- */
-public class PEntryObject {
-    @SerializedName(value = "i")
-    protected long id;
-
-    protected PEntryObject(long id) {
-        this.id = id;
-    }
-
-    @Override
-    public int hashCode() {
-        return Objects.hash(id);
-    }
-
-    @Override
-    public boolean equals(Object obj) {
-        if (!(obj instanceof PEntryObject)) {
-            return false;
-        }
-        PEntryObject other = (PEntryObject) obj;
-        return other.id == this.id;
-    }
+ * Each `GRANT`/`REVOKE` statement will create some privilege entry object
+ * For example, `GRANT SELECT ON TABLE db1.tbl1 TO user1` will create an object like (db1, tbl1)
+ * Thus when user1 executes `SELECT * FROM db1.tbl1`, `PrivilegeChecker` will try to find a matching one in all the table
+ * object and check if SELECT action is granted on it.
+ * If the GRANT statement contains `ALL`, for example, `GRANT SELECT ON ALL TABLES IN DATABASE db1 TO user1`, a slightly
+ * different object like (db1, ALL) will be created to represent fuzzy matching.
+ * Another scenario that will take usage of this fuzzy matching is when user1 execute statement like `USE DATABASE db1`.
+ * We shall create a (db1, ALL) to check if there's any table in the database db1 that user1 have any privilege on.
+ *
+ * The matching rule on the above description can be simplified as this:
+ * (db1, tbl1) matches (db1, tbl1)
+ * (db1, ALL) matches (db1, tbl1)
```

Comment:
This rule only applies for `USE DATABASE xxx` senario, right?

Reply:
As well as `SHOW DATABASE` and `SHOW TABLES`. I think that's all, but we will probably need to go through all the statements to ensure.